src/lerna_manifest.rs

```diff
@@ -90,11 +90,11 @@ impl LernaManifest {
     }
 
     pub fn into_package_manifests_by_package_name(
-        mut self,
+        self,
     ) -> Result<HashMap<String, PackageManifest>, Box<dyn Error>> {
         self.internal_package_manifests
-            .drain(0..)
-            .map(|manifest| Ok((manifest.contents.name.to_owned(), manifest)))
+            .into_iter()
+            .map(|manifest| Ok((manifest.contents.name.clone(), manifest)))
```

Comment:
What's the thinking behind this diff?

Reply:
`ToOwned` (and its associated function, `to_owned`) is a generalization of `Clone` so that you can basically clone into another type: for example, you could do
```rust
let x: &str = "wow!";
let owned: String = x.to_owned();
```
But we're not actually using the generalized behavior of `to_owned` here: all it was doing (prior to the diff) was cloning from `&String` to `String`. My thoughts are that it would be better to represent this explicitly: we're just cloning the string simply, not trying to clone it into a `Box` or `Rc` or something.

Reply:
I see, favoring specificity to narrow the range of intent. I wasn't solid on the relationship between `to_owned` and `clone`, thanks for explaining!